### ISSUE TITLE
Improve JVM agent view with per-pool stats

### ISSUE THREAD

**Issue by henrikno** [elastic/apm-agent-java#1790]
## Is your feature request related to a problem?
The current java APM UI can be confusing to read if you don't know the inner workings of the JVM. Specifically, it looks like this heap is almost full all the time:
![image](https://user-images.githubusercontent.com/147660/116279988-14476900-a73d-11eb-9d51-b235f1ef428f.png)

If you split it up per pool it looks better:
![image](https://user-images.githubusercontent.com/147660/116280570-ab142580-a73d-11eb-958c-a0d5512fac26.png)

But it's not until it does a full GC until you know how much memory it's actually using.
![image](https://user-images.githubusercontent.com/147660/116281787-e4996080-a73e-11eb-9d55-ca8e8249a387.png)

## Describe the solution you'd like

I want the visualization to split up per pool (or different charts for old/young gen, depending).

Related ask:
It would be great if we could also record the "minimum" heap size per pool, e.g. what it was after doing a GC. The JVM collects this in 
https://docs.oracle.com/javase/6/docs/api/java/lang/management/MemoryPoolMXBean.html#getCollectionUsage()
This number only makes sense after the first full GC, so it might say that it's using 2MB in the beginning because it did an initial GC during startup. Also not sure how best to visualize this, maybe a dotted line as a "baseline" or something, or separate chart.

## Describe alternatives you've considered

We can collect some of this info with JMX, e.g
`capture_jmx_metrics=object_name[java.lang:type=MemoryPool,name=*] attribute[CollectionUsage:metric_name=collection_usage]` but we have to make custom dashboards, and if you look at the built-in ones it's difficult to know which one is "correct".

**Comment by jackshirazi** [comment]
Available in 8.12
![image](https://github.com/elastic/apm-agent-java/assets/11161990/8ac9ecf6-36a0-4769-bb46-1bca68698c66)
